### ISSUE TITLE
atl_res_patch and atl_check_data flexibility to assign new column names for x and y

### ISSUE THREAD

**Issue by CBeardsworth** [pratikunterwegs/atlastools#29]
I note that atl_check_data is throwing an error because I have nothing called "x" in my df but it is called "X".

**Comment by pratikunterwegs** [comment]
So this issue stems from some opinionated variable naming, where I expect the coordinates to be named `x` and `y`.
Inside `atl_res_patch`, `atl_check_data` is hardcoded to expect `x`, `y`, and `time`.

In other functions, I offer the choice to specify the coordinate (and time) column names.
I didn't do this here because then `atl_res_patch` would have _a lot_ of arguments, which might confuse users.

So the question is, would you prefer:

1. Allowing users to specify the coordinate (and time) columns, at the risk of making too many function arguments in `atl_res_patch`, or

2. Should I recommend somewhere in the documentation of `atl_res_patch` that coordinate names are fixed and users should rename their coordinates?

**Comment by CBeardsworth** [comment]
My go to solution was to first try to name the variables as is the case in atl_get_speed. I later had to change the names but since atl_check_data isn't in the normal pipeline for a user I do not think it is a problem to include them as arguments. So far I have used atl_get_speed, atl_median_smooth and atl_res_patch. for both other functions users are able to define x, y and time.

**Comment by pratikunterwegs** [comment]
Fair enough, I'll work on adding these options.

**Comment by pratikunterwegs** [comment]
I looked into `atl_res_patch`. There are multiple instances where coordinate and time summary variables are used:

Here's an example where the distance between proto-patches is calculated. Note the `x_end`, `y_start` and similar.

      patch_summary[, spat_diff :=
        c(atl_patch_dist(
          data = patch_summary,
          x1 = "x_end", x2 = "x_start",
          y1 = "y_end", y2 = "y_start"
        ))]

Allowing users to keep their column names for coordinates and time would need this changed to flexibly reflect the coordinate names given by users. Why? Because in a previous step, a summary function from `data.table` automatically chooses names for summary variables, such as `x_start`.

The example is easily solved by renaming using `sprintf`. But downstream calculations using `x_start`, `x_end` are not as straightforward to rewrite.

Basically allowing users to choose their own column names is lots of work for me, and I'd prefer to make users do the work of renaming their columns. :)

If there's significant interest in having this option, users are welcome to make a fork and a pull request with the proposed change.
